go/enclave/storage/enclavedb/batch.go

```diff
@@ -195,7 +195,7 @@ func ReadCurrentSequencerNo(db *sql.DB) (*big.Int, error) {
 }
 
 func ReadHeadBatchForBlock(db *sql.DB, l1Hash common.L1BlockHash) (*core.Batch, error) {
-	query := " where is_canonical and b.height=(select max(b1.height) from batch b1 where b1.is_canonical and b1.l1_proof=?)"
+	query := " where is_canonical and b.height=(select max(b1.height) from batch b1 where b1.is_canonical and b1.executed and b1.l1_proof=?)"
```

Comment:
"where is_canonical and executed and b.height=(select max(b1.height) from batch b1 where b1.is_canonical and b1.executed and b1.l1_proof=?)"

To avoid any weird edge case

Reply:
@BedrockSquirrel , while you're here, maybe you can rename the field from "executed" to "is_executed", for consistency.

Reply:
> To avoid any weird edge case

Good shout, didn't notice I'd done that haha